core/tests/net/tcp.cpp

```diff
@@ -0,0 +1,98 @@
+#include <cstring>
+#include <thread>
+
+#include <doctest/doctest.h>
+
+#include <cubos/core/net/address.hpp>
+#include <cubos/core/net/tcp_listener.hpp>
+#include <cubos/core/net/tcp_stream.hpp>
+
+TEST_CASE("tcp")
+{
+    using cubos::core::net::Address;
+    using cubos::core::net::TcpListener;
+    using cubos::core::net::TcpStream;
+
+    SUBCASE("write and read with stream and listener")
+    {
+        auto svAddr = Address::fromIPv4(127, 0, 0, 1);
+        uint16_t svPort = 8080;
+
+        std::thread serverThread{[&svAddr, &svPort]() {
+            TcpListener listener;
+            REQUIRE(listener.listen(svAddr, svPort, 1));
+
+            TcpStream stream;
+            REQUIRE(listener.accept(stream));
+
+            char readBuffer[4 + 1];
+            REQUIRE(stream.read(readBuffer, sizeof(readBuffer)) > 0);
```

Comment:
Two issues here:
- you're reading 5 bytes, `sizeof(readBuffer)`, while only 4 are being sent. 
- sockets reads are not guaranteed to return the entire requested data (I might be wrong though, if you can, check if that's really the case for blocking sockets). You should repeat `read` calls until all of the buffer has been received. Btw, this is usually necessary because if a signal is triggered, or the OS steals the app context, your read is interrupted. If you're unlucky with the timing, you might read only part of the data.

Reply:
- How should I handle the null termination byte? Declare a buffer of 5 bytes and read 4, append 0 to the last one?
- Is this the client's job to do? Or should we do it in `TcpStream::read`

Reply:
- Yes, that's way to do it.
- Yes it is, although it becomes a bit repetitive, but that's the `Stream::read` interface. If you want to simplify the user code, we can add a `Stream::readAtLeast` (if you have a better name idea feel free to pick something else), that calls `Stream::read` until the requested bytes are received.

Reply:
`readExact` sounds good?

Reply:
LGTM